### PR TITLE
repart: Use crypt_reencrypt_run() if available

### PR DESCRIPTION
**Pull request by daandemeyer**
crypt_reencrypt() is deprecated, so let's look for and prefer crypt_reencrypt_run() if it is available.

### CODE REVIEW

File: src/shared/cryptsetup-util.c [hunk 1]
```diff
@@ -304,6 +307,9 @@ int dlopen_cryptsetup(void) {
 #if HAVE_CRYPT_REENCRYPT_INIT_BY_PASSPHRASE
                         DLSYM_ARG(crypt_reencrypt_init_by_passphrase),
 #endif
+#if HAVE_CRYPT_REENCRYPT_RUN
+                        DLSYM_ARG(crypt_reencrypt_run),
+#endif
 #if HAVE_CRYPT_REENCRYPT
```
Comment (yuwata) [inline src/shared/cryptsetup-util.c]: Maybe, better to load the old one only when the new one is not detected? Then, we can drop the suppression of deprecated warning.

**Review by yuwata — APPROVED** [review]
LGTM. Is this necessary for v256?

**Comment by DaanDeMeyer** [comment]
> LGTM. Is this necessary for v256?

It was causing CI failures in #33146 when building with Werror but I can also try a more minimal check there by just disabling the deprecation warnings when checking for features.

**Comment by DaanDeMeyer** [comment]
I put in a more minimal fix into #33146 instead so this can wait until next release.

**Pull request merged**